File: Ler lista java/clima.java
public class clima {
    public String ano;
    public String mes;
    public String temperatura;
    public String precipitacao;
    
    public clima(String ano, String mes, String temperatura, String precipitacao) {
        this.ano = ano;
        this.mes = mes;
        this.temperatura = temperatura;
        this.precipitacao = precipitacao;
    }

    //reescrevos toString para exibir um objeto completo da classe
    @Override
    public String toString() {
        return "clima [ano=" + ano + ", mes=" + mes + ", temperatura=" + temperatura + ", precipitacao=" + precipitacao + "]";
    }

    
    //reesrevemos equals para definir os atributos identificadores
    @Override
    public boolean equals(Object obj) {
        clima c = (clima)obj;
       return (this.ano.equals(c.ano) && this.mes.equals(c.mes));
    }
}

File: clima python/clima.py
class clima:
    def __init__(self, ano, mes, temperatura, precipitacao):

        self.ano = ano
        self.mes = mes
        self.temmpertaura = temperatura
        self.precipitacao = precipitacao


    def __str__(self):
        return f"Dados climaticos(Ano:{self.ano}. Mes:{self.mes}. Temperatura:{self.temmpertaura}. Precipitacao:{self.precipitacao}"
    
    def __eq__(self, object):
        
        if not isinstance(object, self.__class__):
            return NotImplemented

        return self.ano == object.ano and self.mes == object.mes